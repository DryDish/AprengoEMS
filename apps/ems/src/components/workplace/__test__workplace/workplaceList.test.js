import { render, cleanup } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';

import '@testing-library/jest-dom/extend-expect.js';
import renderer from 'react-test-renderer';
import WorkplaceList from "../WorkplaceList.component";

afterEach(cleanup);

it('Renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<WorkplaceList />, div);
});

it('Matches the snapshot', () => {
    const workplaceListSnapshot = renderer.create(<WorkplaceList/>).toJSON();
    expect(workplaceListSnapshot).toMatchSnapshot();
});

it('Renders Workplace List correctly', () => {
    const {getByTestId} = render(<WorkplaceList/>);
    expect(getByTestId("workplaceList"))
        .toHaveTextContent("Workplace List"
            && "ID"
            && "Name"
            && "Address");
});