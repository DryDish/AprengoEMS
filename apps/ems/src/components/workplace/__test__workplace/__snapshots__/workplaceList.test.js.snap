// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  className="col-"
  data-testid="workplaceList"
  style={
    Object {
      "margin": "auto",
      "width": "80%",
    }
  }
>
  <h3
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Workplace List
  </h3>
  <br />
  <div
    className="container"
  >
    <div
      className="row"
      style={
        Object {
          "fontWeight": "Bold",
        }
      }
    >
      <div
        className="col-1"
      >
        ID
      </div>
      <div
        className="col"
      >
        Name
      </div>
      <div
        className="col-2"
      >
        Address
      </div>
    </div>
  </div>
  <hr />
  <ul
    className="list-group"
  />
</div>
`;
